utils/utils.go

```diff
@@ -224,3 +227,114 @@ func MovePauseProcessToScope(pausePidPath string) {
+func TransferRootful(source entities.ImageScpOptions, dest entities.ImageScpOptions, podman string, parentFlags []string) error {
+	saveCommand := []string{"save", "--output", source.File, source.Image}
+	loadCommand := []string{"load", "--input", dest.File}
+	// if executing using sudo or transferring between two users, the TransferRootless approach will not work, default to using machinectl or su as necessary.
+	// the approach using sudo is preferable and more straightforward. There is no reason for using sudo in these situations
+	// since the feature is meant to transfer from root to rootless an vice versa without explicit sudo evocaiton.
+	var uSave *user.User
+	var uLoad *user.User
+	var err error
+	source.User = strings.Split(source.User, ":")[0] // split in case provided with uid:gid
+	dest.User = strings.Split(dest.User, ":")[0]
+	uSave, err = lookupUser(source.User)
+	if err != nil {
+		return err
+	}
+	switch {
+	case dest.User != "": // if we are given a destination user, check that first
+		uLoad, err = lookupUser(dest.User)
+		if err != nil {
+			return err
+		}
+	case uSave.Name != "root": // else if we have no destination user, and source is not root that means we should be root
+		uLoad, err = user.LookupId("0")
+		if err != nil {
+			return err
+		}
+	default: // else if we have no dest user, and source user IS root, we want to be the default user.
+		uString := os.Getenv("SUDO_USER")
+		if uString == "" {
+			uLoad, err = user.LookupId("1000")
+			if err != nil {
+				return err
+			}
+		} else {
+			uLoad, err = user.Lookup(uString)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	machinectl, err := exec.LookPath("machinectl")
+	if err != nil {
+		logrus.Warn("defaulting to su since machinectl is not available, su will fail if no user session is available")
+		err = execSu(uSave, podman, parentFlags, saveCommand)
+		if err != nil {
+			return err
+		}
+		return execSu(uLoad, podman, parentFlags, loadCommand)
+	}
+	err = execMachine(uSave, podman, parentFlags, saveCommand, machinectl)
+	if err != nil {
+		return err
+	}
+	return execMachine(uLoad, podman, parentFlags, loadCommand, machinectl)
+}
+
+func lookupUser(u string) (*user.User, error) {
+	if u, err := user.LookupId(u); err == nil {
+		return u, nil
+	}
+	return user.Lookup(u)
+}
+
+func execSu(execUser *user.User, podman string, parent, command []string) error {
+	cmd := exec.Command("su", "-l", execUser.Username, "--command")
+	cmdAppend := util.CreateSCPCommand(podman, parent, command, false)
+	cmd.Args = append(cmd.Args, cmdAppend.Args...)
+	logrus.Debug("Executing command su")
+	return cmd.Run()
+}
+
+func execMachine(execUser *user.User, podman string, parent, command []string, machinectl string) error {
+	var cmd *exec.Cmd
+	if execUser.Uid == "0" {
+		cmd = exec.Command("sudo", machinectl, "shell", "-q", execUser.Username+"@.host")
```

Comment:
This looks backwards? If I am root, then I do not need to execute sudo?

Reply:
changed to be an error, if sudo user is not defined there is no way to get the rootless user executing the sudo process.

Reply:
for some reason running as root and executing machinectl w/out using sudo does nothing. I will test this again to be sure